Automattic/woocommerce-services
# Make sure plugin is compatible with WooCommerce 3.0

Closes #653

I've tested this plugin against 2.6 and 2.7, and tried various things like: Getting rates at checkout, printing labels, etc. More testing would be great though.

- Use get_id() instead of ->id because it's become private in 3.0
- Fixed some phpDocs with typos and missing parameters
- Fixed some minor php issues with undefined variables
- add some phpDoc to clarify expected variable types
- Add logic to correctly use $order->id vs. $ $order->get_id();
- Pass order_id to get_form_data to avoid duplicate order_id statement
- Use PHPStorms code analysis to correct various issues
- Add a compatibility class for managing different versions of WC

To test: 
- Try out this branch with the latest WooCommerce 3.0
- Also make sure nothing is broken in previous versions of WooCommerce
